examples/advanced_operations/add_dynamic_search_ads_campaign.py

```diff
@@ -0,0 +1,273 @@
+
+    # Display the results
+    print('Created campaign with resource_name: {}'.format(resource_name))
+
+    return resource_name
+
+
+def create_ad_group(client, customer_id, campaign_resource_name):
+    """Creates a Dynamic Search Ad Group under the given Campaign.
+
+    Args:
+        client: an initialized GoogleAdsClient instance.
+        customer_id: a client customer ID str.
+        campaign_resource_name: a resource_name str for a Campaign.
+
+    Returns:
+        A resource_name str for the newly created Ad Group.
+    """
+    # Retrieve a new ad group operation object.
+    ad_group_operation = client.get_type('AdGroupOperation', version='v1')
+    # Create an ad group.
+    ad_group = ad_group_operation.create
+    # Required: set the ad group's type to Dynamic Search Ads.
+    ad_group.type = client.get_type('AdGroupTypeEnum',
+                                    version='v1').SEARCH_DYNAMIC_ADS
+    ad_group.name.value = 'Earth to Mars Cruises {}'.format(uuid4())
+    ad_group.campaign.value = campaign_resource_name
+    ad_group.status = client.get_type('AdGroupStatusEnum', version='v1').PAUSED
+    # Recommended: set a tracking URL template for your ad group if you want to
+    # use URL tracking software.
+    ad_group.tracking_url_template.value = (
+        'http://tracker.example.com/traveltracker/{escapedlpurl}')
+    # Set the ad group bids.
+    ad_group.cpc_bid_micros.value = 3000000
+
+    # Retrieve the ad group service.
+    ad_group_service = client.get_service('AdGroupService', version='v1')
+    # Submit the ad group ad operation to add an ad group.
+    response = ad_group_service.mutate_ad_groups(customer_id,
+                                                 [ad_group_operation])
+    resource_name = response.results[0].resource_name
+
+    # Display the results.
+    print('Created Ad Group with resource_name: {}'.format(resource_name))
+
+    return resource_name
+
+
+def create_expanded_dsa(client, customer_id, ad_group_resource_name):
+    """Creates a Dynamic Search Ad under the given Ad Group.
+
+    Args:
+        client: an initialized GoogleAdsClient instance.
+        customer_id: a client customer ID str.
+        campaign_resource_name: a resource_name str for an Ad Group.
+    """
+    # Retrieve a new ad group ad operation object.
+    ad_group_ad_operation = client.get_type('AdGroupAdOperation', version='v1')
+    # Create and Expanded Dynamic Search Ad. This ad will have its headline,
+    # display URL and final URL auto-generated at serving time according to
+    # domain name specific information provided by DynamicSearchAdSetting at
+    # the campaign level.
+    ad_group_ad = ad_group_ad_operation.create
+    # Optional: set the ad status.
+    ad_group_ad.status = client.get_type('AdGroupAdStatusEnum',
+                                         version='v1').PAUSED
+    # Set the ad description.
+    ad_group_ad.ad.expanded_dynamic_search_ad.description.value = '''
```

Comment:
This will introduce a \n and extra space in the description - not what you'd want

Reply:
Fixed